src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -179,7 +198,28 @@ void ProbeDirectory(string manifestDirectory)
             {
                 foreach (var kvp in _workloadSet.ManifestVersions)
                 {
-                    manifestIdsToDirectories[kvp.Key.ToString()] = GetManifestDirectoryFromSpecifier(new ManifestSpecifier(kvp.Key, kvp.Value.Version, kvp.Value.FeatureBand));
+                    var manifestSpecifier = new ManifestSpecifier(kvp.Key, kvp.Value.Version, kvp.Value.FeatureBand);
+                    var manifestDirectory = GetManifestDirectoryFromSpecifier(manifestSpecifier);
+                    if (manifestDirectory == null)
+                    {
+                        throw new FileNotFoundException(string.Format(Strings.ManifestFromWorkloadSetNotFound, manifestSpecifier.ToString(), _workloadSet.Version));
+                    }
+                    manifestIdsToDirectories[kvp.Key.ToString()] = manifestDirectory;
+                }
+            }
+
+            //  Load manifests from install state
+            if (_manifestsFromInstallState != null)
+            {
+                foreach (var kvp in _manifestsFromInstallState.ManifestVersions)
+                {
+                    var manifestSpecifier = new ManifestSpecifier(kvp.Key, kvp.Value.Version, kvp.Value.FeatureBand);
+                    var manifestDirectory = GetManifestDirectoryFromSpecifier(manifestSpecifier);
+                    if (manifestDirectory == null)
+                    {
+                        throw new FileNotFoundException(string.Format(Strings.ManifestFromInstallStateNotFound, manifestSpecifier.ToString(), _installStateFilePath));
```

Comment:
I'm wondering if this throw is too aggressive...there are several manifests in this folder, and we only care about one, right? So if an irrelevant manifest is broken, do we actually care? Or are these already scoped down to just the manifests we care about?

Reply:
It's already scoped down.  `_manifestsFromInstallState.ManifestVersions` is only the manifests that were specified in the `manifests` section of global.json, so we need to load each of those.